### PR TITLE
docs: update CICD models' description

### PR DESCRIPTION
**Pull request by Startrekzky**
### ⚠️ &nbsp;&nbsp;Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have `npm run build` and `npm run serve` locally before submitting this PR
- [x] I have read through the [Contributing](https://devlake.apache.org/community/) Documentation

# Summary
Update the description of the records in the CI/CD domain tables. For instance:
1. Add Bamoo entities as examples
2. Update the description for fields with hidden(default) data conversion, e.g. `type`, `status`, `type`, `environment`, etc.
3. Fix the wrong description in table.cicd_pipelines
4. Clarify the description of `prev_success_deployment_commit_id` in table.cicd_deployment_commits
5. and many other changes

All changes are made on main, v0.19 and v0.18 docs.

### Does this close any open issues?
N/A

### Screenshots
N/A

### CODE REVIEW

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 1]
```diff
@@ -441,36 +441,37 @@ Normal comments, review bodies, reviews' inline comments of GitHub's pull reques
...
-| `environment`   | varchar         | 255        | To indicate the environment in which the task is running                                      |                   |
+| `result`        | varchar         | 100        | The result of the pipeline. It will be standardized to 'SUCCESS', 'FAILURE' in DevLake based on each plugin's possible results.                                                |                   |
+| `status`        | varchar         | 100        | The status of the pipeline. It will be standardized to 'DONE', 'IN_PROGRESS' in DevLake based on each plugin's possible statues.      |
+| `type`          | varchar         | 100        | The value will be set to 'DEPLOYMENT' if it matched the ReGex configured in the Scope Config, otherwise it is an empty string.                                                            |                   |
+| `environment`   | varchar         | 255        | The value will be set to 'PRODUCTION' if it matched the ReGex configured in the Scope Config, otherwise it is an empty string.                                      |                   |
```
Comment (d4x1) [inline docs/DataModels/DevLakeDomainLayerSchema.md]: I think `ReGex` should be `regex`, https://en.wikipedia.org/wiki/Regular_expression

File: docs/DataModels/DevLakeDomainLayerSchema.md [hunk 2]
```diff
@@ -441,36 +441,37 @@ Normal comments, review bodies, reviews' inline comments of GitHub's pull reques
...
+| `duration_sec`  | bigint unsigned |            | how long does this pipeline take                                                                  |                   |
+| `created_date`  | datetime        | 3          | When this pipeline started                                                                      |                   |
```
Comment (d4x1) [inline docs/DataModels/DevLakeDomainLayerSchema.md]: `When this pipeline started`  or   `When this pipeline created`?

File: docs/DataModels/DevLakeDomainLayerSchema.md (hunk 1 above)
Comment (Startrekzky) [reply docs/DataModels/DevLakeDomainLayerSchema.md]: @d4x1 I was trying to use RegEx...

File: docs/DataModels/DevLakeDomainLayerSchema.md (hunk 2 above)
Comment (Startrekzky) [reply docs/DataModels/DevLakeDomainLayerSchema.md]: They are the same in this case, but I can get it updated.

**Review by d4x1 — APPROVED** [review]

**Pull request merged**